File: src/main/java/com/github/highright1234/glacier/protocol/packet/status/server/SLPResponse.java
package com.github.highright1234.glacier.protocol.packet.status.server;

import com.github.highright1234.glacier.protocol.AbstractPacket;
import io.netty.buffer.ByteBuf;
import lombok.Data;
import lombok.EqualsAndHashCode;

@EqualsAndHashCode(callSuper = true)
@Data
public class SLPResponse extends AbstractPacket {

    private String jsonResponse;

    @Override
    public void write(ByteBuf buf) {
        writeString(jsonResponse, buf);
    }
}

File: src/main/kotlin/com/github/highright1234/glacier/protocol/packet/status/server/SLPResponse.kt
package com.github.highright1234.glacier.protocol.packet.status.server

import com.github.highright1234.glacier.SLPResponseData
import com.github.highright1234.glacier.protocol.MinecraftPacket
import com.github.highright1234.glacier.protocol.readString
import com.google.gson.Gson
import kotlin.Throws
import java.lang.Exception
import io.netty.buffer.ByteBuf

data class SLPResponse(var slpResponseData : SLPResponseData) : MinecraftPacket() {

    override fun write(buf: ByteBuf) {
        writeString(Gson().toJson(slpResponseData), buf)
    }

    @Throws(Exception::class)
    override fun read(buf: ByteBuf) {
        slpResponseData = Gson().fromJson(buf.readString(), SLPResponseData::class.java)
    }
}